### PR TITLE
Add refreshable configuration abstract classes

### PR DESCRIPTION
**Pull request by sergeysenja1992**

### CODE REVIEW

File: xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/api/refreshable/AbstractRefreshableConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+public abstract class AbstractRefreshableConfiguration<CONFIG, CONFIG_FILE> implements RefreshableConfiguration {
+
+    private static final String TENANT_NAME = "tenantName";
+
+    protected final Map<String, CONFIG> configurationsByTenant = new ConcurrentHashMap<>();
+    protected final Map<String, Map<String, CONFIG_FILE>> configurationsByTenantByFile = new ConcurrentHashMap<>();
```
Comment (sergeysenja1992) [inline xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/api/refreshable/AbstractRefreshableConfiguration.java:25]: add javadoc about that in this map

File: xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/config/InitRefreshableConfigurationBeanPostProcessor.java [hunk 2]
```diff
@@ -19,6 +20,8 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static org.apache.commons.lang3.StringUtils.length;
```
Comment (sergeysenja1992) [inline xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/config/InitRefreshableConfigurationBeanPostProcessor.java:23]: remove unused imports

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/api/Condition.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.icthh.xm.commons.flow.api;
+
+import com.icthh.xm.commons.lep.api.BaseLepContext;
+
+public interface Condition {
+    <T extends BaseLepContext & FlowLepContextFields> Boolean checkCondition(T lepContext);
```
Comment (sergeysenja1992) [inline xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/api/Condition.java:6]: align to Predicate

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/context/TenantResourceLepAdditionalContext.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    private final TenantResourceConfigService resourceService;
+
+    @Override
+    public String additionalContextKey() {
+        return FIELD_NAME;
+    }
+
+    @Override
+    public Map<String, Map<String, TenantResource>> additionalContextValue() {
+        return resourceService.getResourcesByType();
```
Comment (sergeysenja1992) [inline xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/context/TenantResourceLepAdditionalContext.java:27]: return not a TenantResource, return data field

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/TenantResourceConfigService.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        return List.copyOf(configuration.values());
+    }
+
+    public TenantResource getByKey(String resourceKey) {
+        return getConfiguration().get(resourceKey);
+    }
+
+    public Map<String, TenantResourceConfig> removeResource(String resourceKey) {
+        Map<String, TenantResourceConfig> updateFiles = new HashMap<>();
+        getConfigurationFiles().forEach((file, config) -> {
```
Comment (sergeysenja1992) [inline xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/TenantResourceConfigService.java:79]: copy before operation

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/TenantResourceConfigService.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    public List<TenantResource> getByResourceType(String resourceType) {
+        return getConfiguration().values().stream()
+            .filter(resource -> resource.getResourceType().equals(resourceType))
+            .collect(toList());
+    }
+
+    public Map<String, TenantResourceConfig> updateFileConfiguration(TenantResource resource) {
+        Map<String, TenantResourceConfig> configurationFiles = getConfigurationFiles();
```
Comment (sergeysenja1992) [inline xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/TenantResourceConfigService.java:98]: copy before mutation

**Review by amedvedchuk — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]

**Pull request merged**